python/ray/tests/test_resource_demand_scheduler.py

```diff
@@ -1202,6 +1202,52 @@ def test_packing(self):
         assert to_launch == {}
         assert not rem
 
+    def test_skip_placed_bundles(self):
+        # test that we do not launch new nodes for bundles that are already placed.
+        provider = MockProvider()
+        scheduler = ResourceDemandScheduler(
+            provider,
+            TYPES_A,
+            10,
+            head_node_type="p2.8xlarge",
+            upscaling_speed=1,
+        )
+
+        provider.create_node({}, {TAG_RAY_USER_NODE_TYPE: "p2.8xlarge"}, 1)
+        # At this point our cluster has 1 p2.8xlarge instances (8 GPUs) and is
+        # fully idle.
+        nodes = provider.non_terminated_nodes({})
+
+        resource_demands = [{"GPU": 1}] * 4
```

Comment:
Remove this. It implies that 4 GPUs on the p2.8xlarge are occupied by these resource demands, which isn’t easy to understand at first glance.

In addition, we also need to check whether the 4 GPUs on the p2.8xlarge are actually occupied. If they aren’t, and the bundles require 8 GPUs in total, the test may still pass even though the underlying behavior is incorrect.

1. Remove `resource_demands`.
2. Increase each bundle from 2 GPUs to 4 GPUs.